students/viveksyngh/quiz.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"encoding/csv"
+	"os"
+	"fmt"
+	"strings"
+	"log"
+	"flag"
+	"time"
+	"math/rand"
+) 
+
+type Question struct {
+	question string
+	answer string
+}
+
+func getQuestions(filePath string) ([]Question) {
+	file, err := os.Open(filePath)
+	if(err != nil){
+		log.Fatal("Failed to open file.")
+	}
+	reader := csv.NewReader(file)
+	questionList, err := reader.ReadAll()
+	if(err != nil) {
+		log.Fatal("Failed to parse CSV file.")
+	}
+	questions := make([]Question, 0)
+	
+	for _, question := range questionList {
+			questions = append(questions, Question{strings.TrimSpace(question[0]), 
+													strings.TrimSpace(question[1])})
+	}
+	return questions
+}
+
+func Quiz(questions []Question, timer *time.Timer) (score int){
+
+	for i, question := range questions {
+		fmt.Printf("Problem #%d %s : ", i + 1, question.question)
+		answerChannel := make(chan string)
+		go func() {
+			var userAnswer string
+			fmt.Scanln(&userAnswer)
+			answerChannel <- userAnswer
+		}()
+
+		select {
+			case <-timer.C:
+				fmt.Println("\nTimeout")
+				return score
+			case userAnswer := <- answerChannel:
+				if(strings.TrimSpace(userAnswer) == question.answer) {
+					score += 1
+				}
+		}
+	}
+	return score
+}
+
+func randomize(questions []Question) []Question{
+	n := len(questions)
```

Comment:
You can do this too:
```go
rand.Seed(time.Now().UnixNano())
for i := range questions {
	j := rand.Intn(i + 1)
	questions[i], questions[j] = questions[j], questions[i]
}
```
or
```go
rand.Seed(time.Now().UnixNano())
rand.Shuffle(len(questions), func(i, j int) {
	questions[i], questions[j] = questions[j], questions[i]
})
```
It's a good idea to seed your random generator otherwise the shuffle returns the same results every time you run the program, which defeats its purpose.